include/alp/algorithms/symm_tridiag_eigensolver.hpp

```diff
@@ -0,0 +1,542 @@
+/*
+ *   Copyright 2022 Huawei Technologies Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <sstream>
+
+#include <alp.hpp>
+#include <graphblas/utils/iscomplex.hpp> // use from grb
+#ifdef DEBUG
+#include "../tests/utils/print_alp_containers.hpp"
+#endif
+
+// TEMPDISABLE should be removed in the final version
+#define TEMPDISABLE
+
+namespace alp {
+
+	namespace algorithms {
+
+		/**
+		 * find zero of secular equation in interval <a,b>
+		 * using bisection
+		 * this is not an optimal algorithm and there are many
+		 * more efficient implementation
+		 */
+		template<
+			typename D,
+			typename VecView1,
```

Comment:
We have to do this when matrix is a composition of select views and/or other views.

Reply:
I don't think we need them as we do not use view and imf types in the function implementation (it would be wrong if this would happen as those are an internal arguments). So we can simplify the implementation by generalizing over a Vector and Matrix type. Here an example from `get_view`:
```
template<
	typename TargetStructure,
	typename SourceMatrix,
	typename SelectVectorR, typename SelectVectorC,
	std::enable_if_t<
		is_matrix< SourceMatrix >::value &&
		is_vector< SelectVectorR >::value &&
		is_vector< SelectVectorC >::value
	> * = nullptr
>
...
get_view(
	SourceMatrix &source,
	const SelectVectorR &sel_r,
	const SelectVectorC &sel_c
) { ..}
```

Reply:
Fixed.